torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -340,3 +344,59 @@ def transpose_matmul(A: torch.Tensor, B: torch.Tensor, Atrans: bool, Btrans: boo
     if Btrans:
         B = B.transpose(-1, -2)
     return torch.matmul(A, B)
+
+
+_pointwise_ops = [
+    operator.add,
+    operator.sub,
+    operator.mul,
+    operator.iadd,
+    operator.isub,
+    operator.imul,
+    operator.itruediv,
+    operator.ifloordiv,
+]
+
+
+def boolean_mask_rewritten(graph: torch.fx.Graph) -> torch.fx.Graph:
+    for node in graph.nodes:
+        if CallFunctionVarArgs(operator.setitem).match(node):
+            # value = pointwise_op(tensor[index])
+            # tensor[index] = value
+            tensor, index, value = node.args
+            if (
+                isinstance(index, torch.fx.node.Node)
+                and "example_value" in index.meta
+                and isinstance(index.meta["example_value"], FakeTensor)
+                and index.meta["example_value"].dtype == torch.bool
+            ):
+                if CallFunctionVarArgs(_pointwise_ops).match(value):
```

Comment:
I'm not familiar with this matcher API. Is it guaranteed there are no intervening nodes between the getiteim and the setitem? If there can be intervening nodes, this may not be safe, because there may be aliasing mutation going on.

Reply:
We can guarantee, if there is intervening nodes, the match is failed.

Reply:
I think we can also write a function and use ```fx_to_pattern``` to generate a pattern, but I think we may loss a bit flexibility.